omod/src/test/java/org/openmrs/module/metadatamapping/web/rest/MetadataSourceResourceOperationTest.java

```diff
@@ -0,0 +1,177 @@
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+/**
+ * Test different REST api operations on {@link org.openmrs.module.metadatamapping.MetadataSource}.
+ */
+@SuppressWarnings("Duplicates")
+public class MetadataSourceResourceOperationTest extends MainResourceControllerTest {
+	
+	@Before
+	public void setup() throws Exception {
+		executeDataSet("metadataMappingInMemoryTestDataSet.xml");
+	}
+	
+	@Test
+	public void create_shouldPersist() throws Exception {
+		// given
+		SimpleObject postData = new SimpleObject().add("name", "my-source");
+		
+		// when
+		SimpleObject postResponseData = deserialize(handle(newPostRequest(getURI(), postData)));
+		
+		// then
+		assertNotNull(postResponseData);
+		Object newUuid = PropertyUtils.getProperty(postResponseData, "uuid");
+		assertNotNull(newUuid);
+		
+		SimpleObject getResponseData = deserialize(handle(newGetRequest(getURI() + "/" + newUuid)));
+		assertNotNull(getResponseData);
+		assertEquals("my-source", PropertyUtils.getProperty(getResponseData, "name"));
+	}
+	
+	@Test
+	public void update_shouldPersist() throws Exception {
+		// given
+		SimpleObject postData = new SimpleObject().add("description", "This is the new description");
+		
+		// when
+		MockHttpServletResponse postResponse = handle(newPostRequest(getURI() + "/" + getUuid(), postData));
+		
+		// then
+		assertNotNull(postResponse);
+		SimpleObject getResponseData = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertNotNull(getResponseData);
+		assertEquals("This is the new description", PropertyUtils.getProperty(getResponseData, "description"));
+	}
+	
+	@Test
+	public void delete_shouldRetire() throws Exception {
+		// given
+		SimpleObject unRetired = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertEquals(Boolean.FALSE, PropertyUtils.getProperty(unRetired, "retired"));
+		
+		// when
+		MockHttpServletResponse deleteResponse = handle(newDeleteRequest(getURI() + "/" + getUuid()));
+		
+		// then
+		assertNotNull(deleteResponse);
+		SimpleObject retired = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertEquals(Boolean.TRUE, PropertyUtils.getProperty(retired, "retired"));
+	}
+	
+	@Test
+	public void getSingle_shouldGetByName() throws Exception {
+		// given
+		// test data
+		
+		// when
+		// NOTE: Space characters as part of request url may seem dangerous but here we assume
+		// than in an actual http request the client would take of url escaping, if necessary.
+		SimpleObject getResponseData = deserialize(handle(newGetRequest(getURI() + "/Integration Test Metadata Source 2")));
```

Comment:
We allow for fetching by uuid only that way... If you want to find by name then you need to use a query... ?name="Integration Test Metadata Source 2".